src/main/java/pl/cleankod/exchange/core/gateway/AccountService.java

```diff
@@ -0,0 +1,24 @@
+package pl.cleankod.exchange.core.gateway;
+
+import org.springframework.lang.Nullable;
+import pl.cleankod.exchange.core.domain.Account;
+
+import java.util.Optional;
+
+public interface AccountService {
```

Comment:
I see only one implementation of this, which raises the question of why we need this interface in the first place.

Reply:
That one is kind of philosophical / conventions question, I would argue to use interfaces. 

Spring utilizes proxies a lot, when it have an interface it can use JDK proxy, for classes it has to fallback to CGLIB. While functionally it doesn't matter I do like to give a helping hand to the framework when possible, so it gives less errors back to me in return. Jokes aside, subjectively it feels cleaner to me. Helps with testing and future extension.